### PR TITLE
fix: use direct links for logging in

### LINKED ISSUE #3338: The LoginRedirect component causes HTTP 500 during the login flow
Using the `LoginRedirect` component instead of a simple `<a href=...>` causes the browser to issue a request to `/api/data/platform/config` just after the browser starts loading the `/api/auth/login` request. This causes a race condition with sessions and the `Set-Cookie` header, resulting in a failed authentication flow.

1. Request to `/api/auth/login` -> sets the initial cookie value
2. Request to `/api/data/platform/config` -> if it happens before receiving the `Set-Cookie`, will result in a concurrent `Set-Cookie` being received.
3. Browser saves to wrong cookie and uses it during the callback stage of the login flow.

### PR DESCRIPTION
**Pull request by leafty**
Fixes #3338, fixes https://github.com/SwissDataScienceCenter/renku-gateway/issues/729.

Remove the `/login` page (it was only doing a redirect) and replace all links to `/login` with links to `/api/auth/login` which is the login endpoint.

Doing the redirect in-app added unnecessary complexity and caused HTTP 500 errors with the new gateway.

/deploy renku=leafty/update-login-tests

### CODE REVIEW

**Review by andre-code — COMMENTED** [review]
The code looks great 👏! 
There’s one case that isn’t working correctly.

File: client/src/project/components/ProjectViewNotFound.tsx [hunk 1]
```diff
@@ -51,21 +50,14 @@ export const ProjectViewNotFound = ({
...
-          </p>
-        </>
-      );
-    })()
+    <>
+      <p>
+        You might need to be logged in to see this project. Please try to{" "}
+        <Link className="btn btn-secondary btn-sm" to={loginUrl.href}>
+          Log in
+        </Link>
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/project/components/ProjectViewNotFound.tsx:58]: Thanks, I will check again, I thought I replaced all of them.

[the remark this answers is not in the document]
Comment (leafty) [reply client/src/project/components/ProjectViewNotFound.tsx]: Done in 1a24093fb6ed8537815b826c6ea05080f57f9917.

**Review by andre-code — APPROVED** [review]
lgtm 🚀

**Pull request merged**